src/app/hooks/useRequest.js

```diff
@@ -0,0 +1,36 @@
+import { useState, useEffect } from 'react';
+
+import { errorApiHandler } from '../../utils/errorHandler';
+
+const useRequest = ({ request, payload }) => {
+  const [isLoading, setIsLoading] = useState();
+  const [submitError, setSubmitError] = useState([]);
+  const [response, setResponse] = useState();
+
+  const sendRequest = () => {
+    setIsLoading(true);
+    request(payload).then(data => {
+      if (data && data.ok) {
+        setSubmitError([]);
+        // eslint-disable-next-line no-console
+        console.log(data);
+      } else {
+        setSubmitError(errorApiHandler(data));
+      }
+      setIsLoading(false);
+      setResponse(data);
+    });
+  };
+
+  useEffect(() => {
+    if (!payload || !Object.entries(payload).length) {
+      return;
+    }
+
+    sendRequest();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [payload, request]);
+  return [isLoading, submitError, sendRequest, response];
```

Comment:
i would change the order, first loading, then error, then response and last the sendRequest